### DISCUSSION TITLE
Setting up CDN with Digital Ocean

### DISCUSSION THREAD

**Discussion by DidierMalenfant** [owncast/owncast#2322] [Q&A]
I have a working stream with an S3 bucket on digital ocean and my endpoint in `https://nyc3.digitaloceanspaces.com`.

I'm trying to get CDN to work. Currently I'm keeping the endpoint the same, setting the serving endpoint to`https://nyc3.cdn.digitaloceanspaces.com` and enabling CDN on digital ocean.

This breaks the stream with frame rate dropping to something like one frame every 5-10 seconds. Not seeing any errors on the owncast server I don't think. No errors on OBS either.

Any obvious mistakes on my end?

**Comment by gabek** [comment]
What errors do you see in your browser console?

**Reply by DidierMalenfant** [reply]
Looks like it's a CORS error... I get:
```
Blocage d’une requête multiorigine (Cross-Origin Request) : la politique « Same Origin » ne permet pas de consulter la ressource distante située sur https://nyc3.cdn.digitaloceanspaces.com/hls/2/stream.m3u8?cachebust=c1174fd4. Raison : échec de la requête CORS.
```
and then the client tries another playlist, fails with the same error and then loops through all playlists like this.

I only have my owncast URL `https://for.theloveof.house` added to the CORS setting right now but maybe I need a custom domain set up for the CDN so that I can add that to CORS too?

**Reply by gabek** [reply]
Have you been able to work out your CORS issue with DigitalOcean's S3 yet?

**Reply by DidierMalenfant** [reply]
Not yet. I'm still at the error message I'm getting above. I have to learn more about CORS.

My understanding is that it's trying to access something off the CDN url (correctly) but the the browser denies this because it's not coming from the website's own url (https://for.theloveof.house). What I don't understand is why does it work when the source of the stream is the regular (non-CDN) bucket at `https://nyc3.digitaloceanspaces.com`?

**Reply by DidierMalenfant** [reply]
I just tried adding both `https://for.theloveof.house` and `http://for.theloveof.house` to the CORS config on Digital Ocean. No difference. Will keep on investigating.
